components/org.wso2.carbon.identity.media.core/src/main/java/org/wso2/carbon/identity/media/core/StorageSystemManager.java

```diff
@@ -100,77 +107,65 @@ public DataContent readContent(String id, String tenantDomain, String type) thro
                     id, type, tenantDomain));
         }
-        StorageSystemFactory storageSystemFactory = getStorageSystemFactory(StorageSystemUtil.getMediaStoreType());
-        if (storageSystemFactory != null) {
-            return storageSystemFactory.getInstance().isDownloadAllowedForPublicMedia(id, type, tenantDomain);
-        }
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("StorageSystemFactory object is null hence returning security evaluation result as false.");
-        }
-        return false;
+        StorageSystemFactory storageSystemFactory = getStorageSystemFactory(getMediaStoreType());
+        return storageSystemFactory.getInstance().isDownloadAllowedForPublicMedia(id, type, tenantDomain);
     }
 
     /**
      * Security evaluation for downloading protected resource.
      *
-     * @param id           The unique id related to the requesting resource.
+     * @param mediaId      The unique id related to the requesting resource.
      * @param type         The high level content-type of the resource (if media content-type is image/png then
      *                     type would be image).
      * @param tenantDomain The tenant domain of the service call.
+     * @param username     The username of the user.
      * @return true if access to the resource is permitted.
-     * @throws StorageSystemException Exception related to security evaluation during file download.
+     * @throws StorageSystemServerException The server exception related to security evaluation during file download.
      */
-    public boolean isDownloadAllowedForProtectedMedia(String id, String type, String tenantDomain)
-            throws StorageSystemException {
+    public boolean isDownloadAllowedForProtectedMedia(String mediaId, String type, String tenantDomain, String username)
+            throws StorageSystemServerException {
 
         if (LOGGER.isDebugEnabled()) {
             LOGGER.debug(String.format("Evaluate download security for media of type: %s, unique id: %s and tenant " +
-                            "domain %s.", id, type, tenantDomain));
-        }
-        StorageSystemFactory storageSystemFactory = getStorageSystemFactory(StorageSystemUtil.getMediaStoreType());
-        if (storageSystemFactory != null) {
-            return storageSystemFactory.getInstance().isDownloadAllowedForProtectedMedia(id, type, tenantDomain);
-        }
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("StorageSystemFactory object is null hence returning security evaluation result as false.");
+                            "domain %s.", type, mediaId, tenantDomain));
         }
-        return false;
+        StorageSystemFactory storageSystemFactory = getStorageSystemFactory(getMediaStoreType());
+        String userId = getUserIdFromUserName(username);
+        return storageSystemFactory.getInstance().isDownloadAllowedForProtectedMedia(mediaId, type, tenantDomain,
```

Comment:
ideally getStorageSystemFactory should return an instance of a StorageSystemFactory. you need to call isDownloadAllowedForProtectedMedia on that instance. Therefore no need to call getInstance() on an instance (it getInstance is used with singleton class)

Reply:
fixed with 93a57530615d50896104e8d6407e96a73f289b57